lib/perl/Genome/Sys/Lock.pm

```diff
@@ -97,6 +97,8 @@ sub lock_resource {
 
     $class->_cleanup_handler_check();
 
+    $RESOURCE_LOCK_SCOPE{$args{resource_lock}} = $args{scope};
```

Comment:
Yeah, I agree with the sentiment.  I'm glad this PR is generating some discussion about it!